### ISSUE TITLE
config file doesn't parse - attribute error looking up key 'alias'

### ISSUE THREAD

**Issue by rrauenza** [TKCERT/pfFocus#2]
I was about to go write something similar (but more rudimentary...) - thank you for this!

My config file doesn't parse.  I'd rather not post it in case I haven't scrubbed it -- I can send it to you offline if you wish.

First error is in NAT parsing:
```

  File "./format.py", line 73, in <module>
    main()
  File "./format.py", line 68, in main
    step_stdout(args, doc, output_func)
  File "./format.py", line 48, in step_stdout
    output_func(doc, output_file)
  File "/home/rrauenza/src/pfFocus/markdown.py", line 148, in output_markdown
    output_markdown_table(stream, ('Disabled', 'Interface', 'Source', 'Destination', 'Protocol', 'Target', 'Local port', 'Description'), rules)
  File "/home/rrauenza/src/pfFocus/markdown.py", line 73, in output_markdown_table
    stream.write(" | ".join(map(format_markdown_cell, row)))
  File "/home/rrauenza/src/pfFocus/markdown.py", line 36, in format_markdown_cell
    if cell is None or (isinstance(cell, PfSenseNode) and cell.data is None):
  File "/home/rrauenza/src/pfFocus/pfsense.py", line 16, in __getattr__
    return super().__getattribute__(name)
  File "/home/rrauenza/src/pfFocus/util.py", line 13, in data
    data[key] = value.data
  File "/home/rrauenza/src/pfFocus/pfsense.py", line 16, in __getattr__
    return super().__getattribute__(name)
  File "/home/rrauenza/src/pfFocus/pfsense.py", line 86, in data
    for alias in self.rootdoc.pfsense.aliases.alias:
  File "/home/rrauenza/src/pfFocus/pfsense.py", line 16, in __getattr__
    return super().__getattribute__(name)
AttributeError: 'PfSenseAliases' object has no attribute 'alias'
```

I started off just putting in hacky patches to see if it was something simple.
```

--- a/pfsense.py
+++ b/pfsense.py
@@ -11,8 +11,12 @@ class PfSenseNode(DataNode):
         self._parent = parent
 
     def __getattr__(self, name):
         # This trick hides PyLint error messages...
-        return super().__getattribute__(name)
+         This trick hides PyLint error messages...
+        try:
+            return super().__getattribute__(name)
+        except AttributeError:
+            if name == 'alias':
+                return None
```

Which actually made things worse.  Ah!  I think I've walked myself into a fix:

```
@@ -79,9 +83,12 @@ class PfSenseAliasString(PfSenseString):
     @property
     def data(self):
         data = super().data
-        for alias in self.rootdoc.pfsense.aliases.alias:
-            if alias.name.string == data:
-                return {'alias': alias.data}
+        try:
+            for alias in self.rootdoc.pfsense.aliases.alias:
+                if alias.name.string == data:
+                    return {'alias': alias.data}
+        except AttributeError:
+            pass
         return data
 
 class PfSensePortString(PfSenseAliasString):
@@ -136,9 +143,12 @@ class PfSenseRuleAlias(PfSenseString):
             if interface_name == alias_name:
                 interface_data['name'] = data
                 return {'interface': interface_data}
-        for alias in self.rootdoc.pfsense.aliases.alias:
-            if alias.name.string == data:
-                return {'alias': alias.data}
+        try:
+            for alias in self.rootdoc.pfsense.aliases.alias:
+                if alias.name.string == data:
+                    return {'alias': alias.data}
+        except AttributeError:
+            pass
         return data

```

That's kind of ugly -- maybe a if is needed to see if the attribute exists or get('alias', [])  ... or is this just a typo/mistake and you should be iterating instead on self.rootdoc.pfsense.aliases not self.rootdoc.pfsense.aliases**.alias**?

**Comment by mback2k** [comment]
Thanks for the issue report. Please give it another try, I just pushed my version of a first fix.

**Comment by khorsmann** [comment]
Hi Marc,

my merged patch of #6 should also fix that problem too.
So you can close #2 also.

Cheers
Karsten

**Comment by mback2k** [comment]
Hi Karsten, thanks for the info. Yes, your patch should fix this issue as well. Closing this now.
